grpc_api_configuration.yaml

```diff
@@ -5,3 +5,6 @@ http:
   rules:
     - selector: rzp.common.health.v1.HealthCheckAPI.Check
       get: "/v1/healthcheck"
+    - selector: rzp.metro.producer.v1.ProducerApi.Produce
+      post: "/v1/produce"
```

Comment:
Are we going for RESTful APIs? If yes, this is not RESTful, the API should be on top of a resource which would be a noun, produce is an action/verb. `POST /v1/messages` for example fits.

Reply:
Agree. Also, this file is not required now. Use the right annotations in metro-proto itself. Please use this as reference https://github.com/googleapis/googleapis/blob/master/google/pubsub/v1/pubsub.proto and we should be able to copy the same terminology

Reply:
👍

Reply:
Noob here about RPCs in general, but from my understanding, an RPC call is like a Command, so why does it have to be a RESTful API? Isn't `/publish` a better endpoint to have?

And similarly, why is it `ProducerApi.Produce` and not `ProducerApi.Publish`?

Reply:
The reference shared by Abhishek above also is going for a Command paradigm. If I'm reading it correctly, but aren't the two endpoints `.../acknowledge` and `.../pull`?

![image](https://user-images.githubusercontent.com/901108/101357824-fb3e3880-38bf-11eb-9b61-7fb4bd36376e.png)

Reply:
> RPC call is like a Command, so why does it have to be a RESTful API?

You're right, it doesn't. I was saying "if" we're doing REST-ful, let's stick to it's guidelines. If not, RPC-like APIs are good, so `/publish` or `/publishMessage` will fit well.


> The reference shared by Abhishek above also is going for a Command paradigm. If I'm reading it correctly, but aren't the two endpoints .../acknowledge and .../pull?

Google OTOH has a custom spec for representing actions in a REST API, [see here](https://cloud.google.com/apis/design/custom_methods). So you're looking at REST APIs with a command attached, above the default GET/POST/... standard actions. The endpoint is [/v1/subscription:acknowledge](https://cloud.google.com/pubsub/docs/reference/rest/v1/projects.subscriptions/acknowledge).